app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt

```diff
@@ -20,435 +20,292 @@
-import androidx.compose.ui.text.input.TextFieldValue
-import androidx.compose.ui.text.input.getSelectedText
 import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
-import com.wire.android.R
-import com.wire.android.model.Clickable
 import com.wire.android.ui.common.KeyboardHelper
+import com.wire.android.ui.common.SecurityClassificationBanner
 import com.wire.android.ui.common.colorsScheme
-import com.wire.android.ui.home.conversations.mention.MemberItemToMention
-import com.wire.android.ui.home.conversations.model.EditMessageBundle
-import com.wire.android.ui.home.conversations.model.SendMessageBundle
-import com.wire.android.ui.home.conversations.model.UriAsset
-import com.wire.android.ui.home.conversationslist.model.Membership
-import com.wire.android.ui.home.messagecomposer.attachment.AttachmentOptions
-import com.wire.android.ui.home.messagecomposer.state.MessageComposeInputState
-import com.wire.android.ui.home.messagecomposer.state.MessageComposeInputType
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.spacers.VerticalSpace
+import com.wire.android.ui.home.messagecomposer.state.AdditionalOptionSubMenuState
 import com.wire.android.ui.home.messagecomposer.state.MessageComposerState
-import com.wire.android.ui.home.newconversation.model.Contact
-import com.wire.android.ui.theme.wireColorScheme
-import com.wire.android.util.EMPTY
+import com.wire.android.ui.home.messagecomposer.state.MessageCompositionInputSize
+import com.wire.android.ui.home.messagecomposer.state.MessageCompositionInputState
+import com.wire.android.ui.home.messagecomposer.state.MessageBundle
+import com.wire.android.util.ui.KeyboardHeight
 import com.wire.kalium.logic.feature.conversation.InteractionAvailability
 import com.wire.kalium.logic.feature.conversation.SecurityClassificationType
 
 @Composable
 fun MessageComposer(
     messageComposerState: MessageComposerState,
-    messageContent: @Composable () -> Unit,
-    onSendTextMessage: (SendMessageBundle) -> Unit,
-    onSendEditTextMessage: (EditMessageBundle) -> Unit,
-    onMentionMember: (String?) -> Unit,
-    onAttachmentPicked: (UriAsset) -> Unit,
-    isFileSharingEnabled: Boolean,
-    interactionAvailability: InteractionAvailability,
-    securityClassificationType: SecurityClassificationType,
-    membersToMention: List<Contact>,
-    onPingClicked: () -> Unit,
-    onShowSelfDeletionOption: () -> Unit,
-    showSelfDeletingOption: Boolean,
-    tempWritableImageUri: Uri?,
-    tempWritableVideoUri: Uri?
+    messageListContent: @Composable () -> Unit
 ) {
-    BoxWithConstraints {
-        val onSendButtonClicked = remember {
-            {
-                val expireAfter = (messageComposerState.messageComposeInputState as? MessageComposeInputState.Active)?.let {
-                    (it.type as? MessageComposeInputType.SelfDeletingMessage)
-                }?.selfDeletionDuration?.value
+    with(messageComposerState) {
```

Comment:
I moved the check on InteractionAvailibity as the first thing we do, as this has no place inside any Composable or anywhere down the tree,  once we know the interaction availiblity we can stop here and display required COmposable